Grid/serialisation/Hdf5IO.h

```diff
@@ -101,6 +106,90 @@ namespace Grid
   template <>
   void Hdf5Writer::writeDefault(const std::string &s, const std::string &x);
   
+  template <typename U>
+  void Hdf5Writer::writeMultiDim(const std::string &s, const std::vector<size_t> & Dimensions, const U * pDataRowMajor, size_t NumElements)
+  {
+    // Hdf5 needs the dimensions as hsize_t
+    const int rank = static_cast<int>(Dimensions.size());
+    std::vector<hsize_t> dim(rank);
+    for(int i = 0; i < rank; i++)
+      dim[i] = Dimensions[i];
+    // write the entire dataset to file
+    H5NS::DataSpace dataSpace(rank, dim.data());
+
+    size_t DataSize = NumElements * sizeof(U);
```

Comment:
The threshold was conventionally on the number of elements, not with this `size(U)` factor. Why is this changed?

Reply:
My bad! This is now back the way it was.